### PR TITLE
The UHK now only sends out data when it changes

### LINKED ISSUE #72: Media key scancode repetition
On Windows {7,8,10} consumer codes are being repeated (quickly toggled) when relevant keys are pressed. I encountered similar issues earlier and though I fixed them all, but apparently not. I'll look into how Windows sends HID reports, and adjust `UpdateUsbReports()` accordingly.

### LINKED ISSUE #84: Whenever the UHK is plugged in a host machine, Chrome Remote Desktop is blocked from remotely interacting with the machine
Whenever the keyboard is plugged in a host machine, Chrome Remote Desktop is blocked from remotely interacting with the machine.

Let's say I have two machines:
- A desktop with UHK plugged in
- A Chromebook

If I use my Chromebook to connect to my desktop machine remotely using Chrome Remote Desktop,
- The Chromebook will not be able to take control of the mouse and keyboard if the UHK is plugged in
- When the UHK is unplugged while the Remote Desktop session is running, it will allow the Chromebook to take control of the mouse
- When the UHK is plugged again, it will prevent the Chromebook from taking control of the mouse again

Usually, Chrome Remote Desktop prevents the Chromebook from sending inputs to the desktop if the host's keyboard has recently been typed on or when the host's mouse has recently been moved or clicked. Presumably the UHK sends some kind of events that triggers some condition?

Anyways it means that the UHK is not behaving like a traditional mouse or keyboard. When the UHK is at rest, I expect it to behave like a traditional keyboard and mouse at rest. This is a huge problem for me as I don't want to unplug the UHK whenever I leave home.

### LINKED ISSUE #85: Windows: UHK prevents sleep
With the UHK connected, my computer running Windows 10 64-bit won't go to sleep at all. Display won't sleep either so I assume that UHK keeps doing something over the USB as disconnecting it fixes the issue.

### PR DESCRIPTION
**Pull request by Lauszus**
Fixes #72, fixes #84 and fixes #85

### CODE REVIEW

File: right/src/usb_interfaces/usb_interface_mouse.c [hunk 1]
```diff
@@ -20,25 +20,25 @@ void ResetActiveUsbMouseReport(void)
...
+                (uint8_t*)getInactiveUsbMouseReport(), USB_MOUSE_REPORT_LENGTH);
+        IsUsbMouseReportSent = true;
+        UsbMouseActionCounter++;
+    }
+    return status;
 }
 
 usb_status_t UsbMouseCallback(class_handle_t handle, uint32_t event, void *param)
 {
-    UsbMouseActionCounter++;
```
Comment (Lauszus) [inline right/src/usb_interfaces/usb_interface_mouse.c:33]: @mondalaci not sure if this was put here on purpose? But I moved it into usbMouseAction, as this is how it is done in the other interfaces.

Comment (mondalaci) [reply right/src/usb_interfaces/usb_interface_mouse.c:33]: It's just a debug variable that is meant to count the invocations of the mouse callback. It's not critical by any means.

**Comment by Lauszus** [comment]
This affects the mouse, as it moves irregular after this commit.

**Comment by Lauszus** [comment]
https://github.com/UltimateHackingKeyboard/firmware/pull/120#issuecomment-395235978 should be fixed in ad72377f901ece788c54e4168beb422f0a33c2b5.

**Comment by mondalaci** [comment]
@Lauszus You're epic! I can confirm that you've just smashed all the 3 bugs referenced.

MULTIKILL!

**Comment by Lauszus** [comment]
@mondalaci thanks! Note that this might also solve: https://github.com/UltimateHackingKeyboard/firmware/issues/87.

**Comment by Jopie01** [comment]
Can you release a new (beta) version of the firmware, so I can test #87

**Comment by mondalaci** [comment]
@Jopie01 Just did that, and commented #87. Please let's continue this discussion there.

**Pull request merged**